[onnx] classification models export

felixdittrich92 opened the pull request:
Counterpart of #789 for all classifiaction models

- [x] onnx export integrated in classification training script
- [x] classification models onnx export test

Issue:
#789 

Any feedback is welcome :hugs:

felixdittrich92 commented:
@fg-mindee 
Any update ? Or keep it for review after 0.5.1 release ? 😄

charlesmindee commented on references/classification/train_pytorch.py:
```diff
@@ -28,6 +28,27 @@
+        export_params=True, opset_version=13, verbose=False
+    )
+    print(f"Model exported to {name}.onnx")
+    try:
+        from onnxruntime.quantization import quantize_qat
+        quantize_qat(f"{name}.onnx", f'{name}.quant.onnx')
+        print(f"Quantized model saved to {name}.quant.onnx")
+    except ImportError:
+        print("ONNX Runtime is not installed. Skipping quantization.")
```
Wince you put onnx in the requirements, I think we don't need the try except and we can simply import it at the beginning of the file

charlesmindee commented on tests/pytorch/test_models_classification_pt.py:
```diff
@@ -89,3 +94,45 @@ def test_crop_orientation_model(mock_text_box):
+            ort_outs = ort_session.run(['logits'], {'input': dummy_input.numpy()})
+            assert isinstance(ort_outs, list) and len(ort_outs) == 1
+            assert ort_outs[0].shape == (batch_size, *output_size)
+    except RuntimeError:
+        assert False
```
Why do you put a try except in this test ?

felixdittrich92 replied:
Good catch  it's removed :+1:

charlesmindee reviewed (commented):
Thanks for this PR, I think the right place for the export_to_onnx() function might be in the package, in the models/utils/export.py file or something like that, what do you think ?

felixdittrich92 commented:
@charlesmindee ok i have moved it to utils/pytorch ftm :+1:  and also removed both deps we can use torch.onnx and the quantization maybe later add directtly as one line in the training scripts wdyt ?

charlesmindee commented on setup.py:
```diff
@@ -65,6 +65,7 @@
     "pytest>=5.3.2",
     "coverage>=4.5.4",
     "hdf5storage>=0.1.18",
+    "onnxruntime>=1.11.0",
```
Since we removed the onnxruntime dep from the requirements, we may want to remove from the setup file as well

charlesmindee commented on setup.py:
```diff
@@ -137,6 +138,7 @@ def deps_list(*pkgs):
     "coverage",
     "requests",
     "hdf5storage",
+    "onnxruntime",
```
same here

charlesmindee reviewed (approved):
Thanks for this feature !
